src/bin/pg_autoctl/supervisor.c

```diff
@@ -675,12 +1162,6 @@ supervisor_restart_service(Supervisor *supervisor, Service *service, int status)
 	}
 	else
 	{
-		/* exit with a non-zero exit code, and process with shutdown sequence */
-		supervisor->cleanExit = false;
-		supervisor->shutdownSequenceInProgress = true;
-
-		(void) supervisor_stop_other_services(supervisor, service->pid);
-
```

Comment:
This change looks highly suspicious in this PR.

Reply:
It codes the behaviour. The function is responsible for restarting based on the policy. What happens after is not part of this function.